src/posts/introduction-webc.md

```diff
@@ -0,0 +1,236 @@
+
+### Asset bundling
+
+When WebC processes your components and it finds `<style>` or `<script>` tags in them, it can extract those assets into bundles for you to use later. This allows you to collect your assets into a single `<style>` tag inside the page `<head>`, rather than leaving `<style>` tags strewn about your HTML. See the Eleventy docs on WebC for information on [how to include bundled assets in your final markup](https://www.11ty.dev/docs/languages/webc/#css-and-js-(bundler-mode)).
+
+> By default, when using the Eleventy plugin for WebC, asset bundling is turned on. If you are using WebC directly, however, bundling is off by default.
+
+### Scoped styles
+
+One concern you may have about this single file component approach is that the styles from one component may leak into another. In our `site-footer.webc` component, for example, we have styles that apply to all `<footer>` elements; if we use a `<footer>` in any other part of our site, it will pick up that padding.
+
+A common solution to this problem with SFCs is to scope the styles using some computer-generated unique class name. If you add the `webc:scoped` attribute to your style tag, WebC will do this for you.
+
+```diff-html
+<style webc:scoped>
+  :host {
+    padding-block: 3rem;
+    padding-inline: 1rem;
+  }
+
+  :host footer > p {
+    font-family: fantasy;
+  }
+</style>
+```
+
+Here we’ve added the `webc:scoped` attribute to our `<style>` element, and introduced the `:host` pseudo-class to our selectors. WebC now knows to generate a unique class for each `<site-footer>` it processes and to replace `:host` with a class selector matching this generated class name. So our HTML will end up looking something like:
+
+```html
+<site-footer class="blarglargl">
+  <footer>
+    <p>&copy; 2022 Yours Truly</p>
+  </footer>
+</site-footer>
+```
+
+And the styles will be:
+
+```css
+.blarglargl {
+  padding-block: 3rem;
+  padding-inline: 1rem;
+}
+
+.blarglargl footer > p {
+  font-family: fantasy;
+}
+```
+
+## Slots
+
+WebC supports the use of `<slot></slot>` for inserting markup inside your components when they’re used. And as with templates and slots for web components, you can define multiple named slots; something that none of the template languages Eleventy currently supports can do.
+
+To illustrate how this can be useful, let’s create a card component that needs to have a title, some media, and a brief description. We won’t concern ourselves with the CSS for this component, but let’s assume we’re doing some fancy things that require us to place the aformentione title, media, and description inside some wrappers. We can define `a-card.webc` like this:
```

Comment:
Love this example! It really emphasizes how WebC is useful for static components 🙌